pallets/xyk/src/lib.rs

```diff
@@ -660,22 +700,21 @@ impl<T: Config> Pallet<T> {
 		let numerator: U256 = input_reserve_saturated
 			.saturating_mul(buy_amount_saturated)
 			.checked_mul(10000.into())
-			.ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
+			.ok_or(Error::<T>::MathOverflow)?;
 
 		let denominator: U256 = output_reserve_saturated
 			.checked_sub(buy_amount_saturated)
-			.ok_or_else(|| DispatchError::from(Error::<T>::NotEnoughReserve))?
+			.ok_or_else(|| Error::<T>::NotEnoughReserve)?
 			.checked_mul(after_fee_percentage.into())
-			.ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
+			.ok_or(Error::<T>::MathOverflow)?;
```

Comment:
before return type was dispatch error, to create dispatch error you need to call a function `DispatchError::from(` for stuch scenario `ok_or_else` is right thing to do, but now it returns with Error, and to construct error we dont need to call a funciton we can instantiate error directly `Error::<T>::MathOverflow` so there is no point in calling funciton as it will be less efficient.

Reply:
Ah! Got it. Thanks!